driver-core/src/main/com/mongodb/connection/QueryResult.java

```diff
@@ -33,6 +34,7 @@
     private final MongoNamespace namespace;
     private final List<T> results;
     private final long cursorId;
+    private final BsonDocument postBatchResumeToken;
```

Comment:
This feels weird here - I initially thought it would it be cleaner to have a `ChangeStreamQueryResult` - but on consideration that seems a lot of work.

It might be cleaner, not to specialize at all and to optionally store the cursor document in its entirety in the QueryResult.  Then any operations that are interested in that access it and use it should they need.

Reply:
I now store the entire document in `QueryResult`.